pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1022,12 +867,12 @@ func isPodUpdated(oldPod, newPod *v1.Pod) bool {
 }
 
 func (p *PriorityQueue) updateInActiveQueue(logger klog.Logger, oldPod, newPod *v1.Pod, oldPodInfo *framework.QueuedPodInfo) bool {
-	p.activeQLock.Lock()
-	defer p.activeQLock.Unlock()
-	if pInfo, exists := p.activeQ.Get(oldPodInfo); exists {
+	p.activeQ.getLock().Lock()
```

Comment:
Actually, I'm surprised the current impl uses unlocked functions a lot, which I believe we should avoid as much as possible.
For example, here any reason we cannot just create `Update` fn in `activeQueuer`, instead of `unlocked().AddOrUpdate`? Do we have to protect `p.UpdateNominatedPod` with activeQ.lock?

Reply:
I didn't want to change both structure and logic in the same PR to make it easier to review and catch possible errors. I will change this in the next PR.